windows/code/RNDeviceInfoCPP.h

```diff
@@ -35,16 +36,30 @@ namespace winrt::RNDeviceInfoCPP
       provider.Add(L"appName", getAppNameSync());
       provider.Add(L"brand", getBrandSync());
       provider.Add(L"model", getModelSync());
+      provider.Add(L"deviceType", getDeviceTypeSync());
     }
 
     bool isEmulatorHelper(std::string model)
     {
       return std::regex_match(model, std::regex(".*virtual.*", std::regex_constants::icase));
     }
 
-    bool isTabletHelper(std::string os)
+    bool isTabletHelper()
     {
-      return !std::regex_match(os, std::regex(".*windowsphone.*", std::regex_constants::icase));
+      auto view = winrt::Windows::UI::ViewManagement::UIViewSettings::GetForCurrentView();
```

Comment:
Still need to move to UI thread before completing PR.
Else it intermittently fails for being on wrong thread.

Reply:
It 100% needs to be on UI thread.  Not a huge fan of this approach, it does work, but I am open to suggested changes.

Reply:
FYI Mike, would like to give until Monday or Tuesday to make sure a few people can get eyes on most recent changes.

Reply:
Sure thing - and I had never thought about hybrids actually being dynamic with respect to whether they should have a touch-oriented or Keyboard/Mouse type interface put up, but honestly even iPadOS behaves similar now IIRC? And Android may as well. If isTablet is intended to be a packaged-up heuristic it should be dynamic with an event/hook as well but I agree that's a bit of overengineering for what I imagine is an immediate problem.

I would still lobby for either APIs per typical input device (hasKeyboard, hasMouse or similar) or perhaps like react-native-netinfo has done more of an "InputDevice" object where all available input devices and output devices were available. That idea could be extended to "Displays" and "Modems" etc for a more general purpose tree of information vs the current approach with an approaches-infinity number of endpoints.

All to say I'm open to re-shaping things to be useful as react-native itself expands to include more than just "handsets and maybe 1 or 2 info tidbits about other things". This is a foundational library and I want it to really support all devices well

Cheers

Reply:
I did a patch-package for our local problem, so I am in no rush for this PR to go thru.  And blocking on the JS thread isn't great either (As per my most recent change).   I do agree with everything you've said and I do think it's a tough situation.  What are good short term and long term considerations, with what level of change required. 

I agree that longer term, having more dynamic keyboard, mouse, and the dynamic event hook nature is definitely a great idea.

For the isTablet today, perhaps I should just revert back to an earlier PR changeset and leave these proposed changes and dynamic async for a future event/api reshaping PR? Then maybe work on a 2nd PR that adds in the methods for for keyboard and mouse and event hook for dynamicTabletMode? 


I will leave this PR open and also gather some more insights/thoughts this coming week.  

Obviously you get the final say on things, so if there is anything you'd personally specifically want, do let me know and I can do that too.

Reply:
Resolving this comment thread and going back to parent PR for further conversation. I think I was able to address most of the concerns with something I know I am happier with than some of my earlier iterations.